pylsd/bindings/lsd_ctypes.py

```diff
@@ -7,47 +7,16 @@
 
 import ctypes
 import os
-import sys
-
+import glob
 
 def load_lsd_library():
-
-    root_dir = os.path.abspath(os.path.dirname(__file__))
-
-    libnames = ['linux/liblsd.so']
-    libdir = 'lib'
-    if sys.platform == 'win32':
-        if sys.maxsize > 2 ** 32:
-            libnames = ['win32/x64/lsd.dll', 'win32/x64/liblsd.dll']
-        else:
-            libnames = ['win32/x86/lsd.dll', 'win32/x86/liblsd.dll']
-    elif sys.platform == 'darwin':
-        libnames = ['darwin/liblsd.dylib']
-
-    while root_dir != None:
-        for libname in libnames:
-            try:
-                lsdlib = ctypes.cdll[os.path.join(root_dir, libdir, libname)]
-                return lsdlib
-            except Exception as e:
-                pass
-        tmp = os.path.dirname(root_dir)
-        if tmp == root_dir:
-            root_dir = None
-        else:
-            root_dir = tmp
-
-    # if we didn't find the library so far, try loading without
-    # a full path as a last resort
-    for libname in libnames:
-        try:
-            # print "Trying",libname
-            lsdlib = ctypes.cdll[libname]
-            return lsdlib
-        except:
-            pass
-
-    return None
+    # may fail if CWD (via sys.path) contains pylsd/bindings/__init__.py,
+    # but otherwise contains the auto-built library for this platform/installation:
+    lib_name = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'lib.*.so')
```

Comment:
This will break OSX/Windows though, can we keep the fallbacks for non-Linux OS?

Reply:
Why, because of the pathname conventions, or because distutils is not capable of C compilation there? (In the latter case, we would still need to ship prebuilt libs, which is impossible to compute for me on these platforms.)

Reply:
Both, because they have different filename conventions (`*.dll` for win, `.dylib` in OSX IIUC) and because we don't have the platforms (or the tiime) to test building on Win/OSX, so I suggest to keep the prebuilt shared libraries for those platforms and keeping the original logic for them.

Reply:
Oh, right. I forgot about the extensions an that C compilation may still require paying money and installing software on these platforms. 

(BTW what's the point of source distributions and the extension module mechanism, if you cannot use it for _portable_ distributions anyway?)

I am a bit hesitant to go back to the old precompiled libraries, because they don't contain what we need and I cannot reproduce them without these OSes.

But maybe we could add some CI configuration for other platforms that prebuilds wheels for us under the current rules...

Reply:
> But maybe we could add some CI configuration for other platforms that prebuilds wheels for us under the current rules...

I share the desire to Do This Right(TM) but think we should be economic here. The previous version apparently worked on those platforms, neither you nor I work on those platforms, so if clean code in that regard is of importance to the community, somebody might step up to implement it for Win/OSX.

Reply:
> The previous version apparently worked on those platforms

Yes, but unfortunately, it did _not_ contain the changes to the C library (18ba1aa529748214b1fa33641fccd14ac066e725 only has source changes). So unless someone rebuilds those libs for us, we cannot ship new ones.

Being economic in the current situation to me means ignoring these platforms for now. We could at least try to provide a manylinux wheel to PyPI and then call on others to provide more wheels, or set-up said CI later-on ourselves.

BTW, how about these examples:
- [Github actions for multi-platform wheels](https://github.com/pallets/markupsafe/blob/master/.github/workflows/build.yaml) (although I must confess I don't quite understand where the actions themselves are defined – maybe these are predefined Github things?)
- [Travis config for x86/ARM manylinux](https://github.com/msgpack/msgpack-python/blob/main/.travis.yml)
- [Github actions for multi-platform wheels](https://github.com/pyca/cryptography/blob/main/.github/workflows/wheel-builder.yml) (more complex still)

Reply:
> BTW, how about these examples:
> 
> * [Github actions for multi-platform wheels](https://github.com/pallets/markupsafe/blob/master/.github/workflows/build.yaml) (although I must confess I don't quite understand where the actions themselves are defined – maybe these are predefined Github things?)
> * [Github actions for multi-platform wheels](https://github.com/pyca/cryptography/blob/main/.github/workflows/wheel-builder.yml) (more complex still)
> * [Travis config for x86/ARM manylinux](https://github.com/msgpack/msgpack-python/blob/main/.travis.yml)
> * [Travis config for Linux and macOS various compilers](https://github.com/Martinsos/edlib/blob/master/.travis.yml)

I think you can scratch that. Have a look at https://github.com/joerick/cibuildwheel

Reply:
@kba I gave up on the Windows build, because I could not convince the MSVC linker to work. It kept giving me `unresolved external symbol PyInit_lib` (regardless of Python version or architecture). I did compare with various example projects using cibuildwheel, but saw no obvious flaw in my config. Maybe we can fix that later.

Anyway, the CI build now produced an [artifact](https://github.com/bertsky/pylsd/suites/2553220991/artifacts/55526422), which I guess you could download manually and use for local `twine` action. It's just that one time (I hope).

Perhaps in the future the CI should also run the `example` as a smoke test on each platform.

Reply:
Maybe we should also include some flashy badges in the README for the CI and for PyPI?

Reply:
I'm all for flashy badges :)